src/Features/WcPayPromotion/Init.php

```diff
@@ -116,5 +125,73 @@ public static function set_gateway_top_of_list( $ordering ) {
 		}
 		return $ordering;
 	}
+
+	/**
+	 * Get WC Pay promotion spec.
+	 */
+	public static function get_wc_pay_promotion_spec() {
+		$promotions = self::get_promotions();
+		$index      = array_search( 'woocommerce-payments', array_column( $promotions, 'product' ), true );
+
+		if ( false !== $index ) {
+			return $promotions[ $index ];
+		}
+		return false;
+	}
+
+	/**
+	 * Go through the specs and run them.
+	 */
+	public static function get_promotions() {
+		$suggestions = array();
+		$specs       = self::get_specs();
+
+		foreach ( $specs as $spec ) {
+			$suggestion    = EvaluateSuggestion::evaluate( $spec );
```

Comment:
Nice reuse of this class!

Do you think "suggestion" works okay for this use case or is there a better noun that fits the gateway suggestions and promotions?

Reply:
Yeah I think `$suggestion` was a copy from somewhere, I can something more specific.

Reply:
The `EvaluateSuggestion` class is used for payment gateway suggestions, but if we think the code between these is going to be identical, then maybe we rename that class as well.

I guess it depends on whether or not "promotions" and "suggestions" are going to share the same schema and abstractions.